examples/07_compare_encoders.py

```diff
@@ -0,0 +1,289 @@
+and those implemented in *skrub*, that is, the |SimilarityEncoder|,
+the |GapEncoder| and the |MinHashEncoder|.
+
+.. note:
+    Reading :ref:`sphx_glr_auto_examples_01_dirty_categories.py` up to
+    *Performance comparison* is highly recommended before reading this example.
+    It's also recommended to continue reading after finishing this example!
+
+We use the `employee salaries <https://www.openml.org/d/42125>`_ dataset
+for illustration purposes.
+
+
+.. |OneHotEncoder| replace::
+    :class:`~sklearn.preprocessing.OneHotEncoder`
+
+.. |SimilarityEncoder| replace::
+    :class:`~skrub.SimilarityEncoder`
+
+.. |GapEncoder| replace::
+    :class:`~skrub.GapEncoder`
+
+.. |MinHashEncoder| replace::
+    :class:`~skrub.MinHashEncoder`
+"""
+
+###############################################################################
+# Encoding dirty categorical variables
+# ------------------------------------
+#
+# As usual, let's first retrieve the dataset:
+
+from skrub.datasets import fetch_employee_salaries
+
+employee_salaries = fetch_employee_salaries()
+
+employee_salaries.description
+
+###############################################################################
+# Alias X, the descriptions of employees (our input data), and y,
+# the annual salary (our target column):
+
+X = employee_salaries.X
+y = employee_salaries.y
+
+###############################################################################
+# Let's carry out some basic preprocessing:
+
+# Overload `employee_position_title` with `underfilled_job_title`,
+# as the latter gives more accurate job titles when specified
+X["employee_position_title"] = X["underfilled_job_title"].fillna(
+    X["employee_position_title"]
+)
+X.drop(labels=["underfilled_job_title"], axis="columns", inplace=True)
+
+X
+
+###############################################################################
+# As we saw in :ref:`sphx_glr_auto_examples_01_dirty_categories.py`, the
+# column `employee_position_title` is dirty.
+#
+# A classical encoding method is one-hot encoding (in the form of the
+# |OneHotEncoder|): it works by creating a column for each category, and
+# filling it with 1 if the category is present in the sample, and 0 otherwise.
+#
+# Let's visualize that!
+#
+# We'll handpick a set of 10 samples which have a similar job name,
+# that is, the position title contains the word "Fire"
+
+sample = X[X["employee_position_title"].str.contains("Fire")].sample(
+    n=10, random_state=5
+)
+
+# We'll keep only the column we're interested in
+sample = sample[["employee_position_title"]]
+
+sample
+
+###############################################################################
+# Next up: import the |OneHotEncoder| and apply it to our sample:
+
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+
+one_hot = OneHotEncoder(sparse_output=False)
+sample_enc_ohe = one_hot.fit_transform(sample)
+
+# (make it look nice in Jupyter by wrapping the resulting array in a DataFrame)
```

Comment:
I would rather we use the "set_output" method of scikit-learn transformer. Just call "one_hot.set_output(transform="pandas")" above